transcom/mymove
# INT B-21429 and B-21431 combo for populating shipment market code

## [B-21429 Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21429)
## [B-21431 Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21431)

## Summary

We want to be able to track if a shipment is a domestic/international shipment, but we can only do that by checking the addresses under the shipment. If both addresses are CONUS then the `market_code` should be `d`, else it's gonna be `i`.

This does contain code needed for [B-21430](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21430) since the `isOconus` value is necessary to determine the shipment's market code.

This PR does the following:
- migration to populate `market_code` columns
- updates all Swagger definitions for shipments to send back `marketCode` to the UI
- updates all `model_to_payload.go` files to send back `marketCode`
- adds in logic in both shipment creation and update that determines if `marketCode` needs to change by checking the address values
- adds/updates relevant tests

### How to test

1. Fire up yer client & server
2. Check your migrations and the `mto_shipments` table - every value in `market_code` should say `d`
3. Create any shipment as a customer/office user with any CONUS addy
4. Throw in some console logs somewhere that you can see the data
5. You should see the `marketCode` value in the shipment object
6. Now... To test if a shipment changes to `i` you need to be able to enter an OCONUS state (in this case, Alaska)
7. Close down your server, in your `envrc` turn `FEATURE_FLAG_ENABLE_ALASKA` to `true`
8. Run `direnv allow` and fire up your server again
9. Create a shipment with an Alaska address
10. Check your logs, should now see `i` in the `marketCode` value on the shipment

## Screenshots
![Screenshot 2024-10-10 at 1 29 48 PM](https://github.com/user-attachments/assets/f9328860-a70a-4e14-ba3b-be337bcff6d8)
